packages/jsii-python-runtime/src/jsii/_kernel/providers/process.py

```diff
@@ -0,0 +1,225 @@
+    return {"$jsii.byref": value.ref}
+
+
+def ohook(d):
+    if d.keys() == {"$jsii.byref"}:
+        return ObjRef(ref=d["$jsii.byref"])
+    return d
+
+
+class _NodeProcess:
+    def __init__(self):
+        self._serializer = cattr.Converter()
+        self._serializer.register_unstructure_hook(
+            LoadRequest,
+            _with_api_key("load", self._serializer.unstructure_attrs_asdict),
+        )
+        self._serializer.register_unstructure_hook(
+            CreateRequest,
+            _with_api_key("create", self._serializer.unstructure_attrs_asdict),
+        )
+        self._serializer.register_unstructure_hook(
+            DeleteRequest,
+            _with_api_key("del", self._serializer.unstructure_attrs_asdict),
+        )
+        self._serializer.register_unstructure_hook(
+            GetRequest, _with_api_key("get", self._serializer.unstructure_attrs_asdict)
+        )
+        self._serializer.register_unstructure_hook(
+            StaticGetRequest,
+            _with_api_key("sget", self._serializer.unstructure_attrs_asdict),
+        )
+        self._serializer.register_unstructure_hook(
+            SetRequest, _with_api_key("set", self._serializer.unstructure_attrs_asdict)
+        )
+        self._serializer.register_unstructure_hook(
+            StaticSetRequest,
+            _with_api_key("sset", self._serializer.unstructure_attrs_asdict),
+        )
+        self._serializer.register_unstructure_hook(
+            InvokeRequest,
+            _with_api_key("invoke", self._serializer.unstructure_attrs_asdict),
+        )
+        self._serializer.register_unstructure_hook(
+            StaticInvokeRequest,
+            _with_api_key("sinvoke", self._serializer.unstructure_attrs_asdict),
+        )
+        self._serializer.register_unstructure_hook(
+            StatsRequest,
+            _with_api_key("stats", self._serializer.unstructure_attrs_asdict),
+        )
+        self._serializer.register_unstructure_hook(ObjRef, _unstructure_ref)
+        self._serializer.register_structure_hook(ObjRef, _with_reference)
+
+    def __del__(self):
+        self.stop()
+
+    def _next_message(self) -> Mapping[Any, Any]:
+        return json.loads(self._process.stdout.readline(), object_hook=ohook)
+
+    def start(self):
+        self._process = subprocess.Popen(
+            "jsii-runtime", shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE
+        )
+        self.handshake()
+
+    def stop(self):
+        # TODO: We can write an empty string here instead?
+        self._process.terminate()
+
+        try:
+            self._process.wait(timeout=5)
+        except subprocess.TimeoutExpired:
+            self._process.kill()
+
+    def handshake(self):
+        resp: _HelloResponse = self._serializer.structure(
+            self._next_message(), _HelloResponse
+        )
+
+        # TODO: Replace with proper error.
+        assert (
+            resp.hello == "jsii-runtime@0.7.4"
```

Comment:
My TODO should probably have been expanded to also include not hardcoding a version number here. This was one of the first bits of code I wrote for this, and I wasn't sure at the time what the versioning semantics were for the jsii-runtime. Ideally this will be a version range specifier (which will entail taking on more dependencies).